src/direct_copy_stubs.c

```diff
@@ -106,6 +109,16 @@ CAMLprim value stub_cleanup(value handle)
   CAMLreturn(Val_unit);
 }
 
+/* Wait for an fd. There will be a subsequent read() or write()
+ * to collect any fd error conditions that might occur */
+static inline int pollwait(int fd, short event) {
+    struct pollfd pfd;
+
+    pfd.fd = fd;
+    pfd.events = event;
+    return poll(&pfd, 1, -1);
```

Comment:
Since we're fixing EINTR below, what if `poll` got an EINTR?

Reply:
Fair point. Can just continue in that case.